### PR TITLE
Use is-object-like for object/fn detection

### PR DESCRIPTION
**Pull request by bendrucker**
Code is unchanged, just have written this a few times and made a module for myself

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -32,5 +32,8 @@
   ],
   "devDependencies": {
     "ava": "0.0.4"
+  },
+  "dependencies": {
```
Comment (sindresorhus) [inline package.json]: Should be placed before devDeps

File: package.json [hunk 2]
```diff
@@ -32,5 +32,8 @@
...
+    "is-object-like": "~1.0.0"
```
Comment (sindresorhus) [inline package.json]: Should use caret `^`

File: index.js [hunk 3]
```diff
@@ -1,8 +1,6 @@
 'use strict';
 
-function isObjOrFn(x) {
-	return (typeof x === 'object' || typeof x === 'function') && x !== null;
-}
+var isObjOrFn = require('is-object-like')
```
Comment (sindresorhus) [inline index.js]: semicolon

**Comment by sindresorhus** [comment]
Sure. But not interested having nested deps for something as simple as this. Instead of depending on `is-object`, just have the exact same function as current inlined here.

**Comment by bendrucker** [comment]
Amended w/ requested changes

**Comment by sindresorhus** [comment]
Awesome :)

**Comment by stevemao** [comment]
Just FYI [is-obj](https://github.com/sindresorhus/is-obj/blob/master/index.js) is exactly the same as [is-object-like](https://github.com/bendrucker/is-object-like/blob/master/index.js) :)

**Comment by sindresorhus** [comment]
Oh, but `is-object-like` is misleading. They're not object-like, they *are* objects. Almost everything in JS are objects.

**Comment by stevemao** [comment]
@bendrucker
`is-object-like` should probably let `null` return `true` as `null` is object-like?

**Comment by bendrucker** [comment]
Per the description, my intent was to cover things that can take arbitrary properties as if they were plain objects

**Pull request merged**